framework/core/js/src/common/states/ModalManagerState.ts

```diff
@@ -19,62 +26,101 @@ export default class ModalManagerState {
   /**
    * @internal
    */
-  modal: null | {
-    componentClass: UnsafeModalClass;
-    attrs?: Record<string, unknown>;
-    key: number;
-  } = null;
+  modal: ModalItem | null = null;
+
+  /**
+   * @internal
+   */
+  modalList: ModalItem[] = [];
 
   /**
    * Used to force re-initialization of modals if a modal
    * is replaced by another of the same type.
    */
   private key = 0;
 
-  private closeTimeout?: NodeJS.Timeout;
+  /**
+   * Shows a modal dialog.
+   *
+   * @deprecated **From Flarum 2.0, stacking will be enabled by default, and values
+   * for `attrs` and `stackModal` will be required.**
+   *
+   * Opening a new modal will close any others currently being shown for backwards
+   * compatibility reasons, until Flarum 2.0.
+   *
+   * @example <caption>Show a modal</caption>
+   * app.modal.show(MyCoolModal, { attr: 'value' });
+   *
+   * @example <caption>Show a modal from a lifecycle method (`oncreate`, `view`, etc.)</caption>
+   * // This "hack" is needed due to quirks with nested redraws in Mithril.
+   * setTimeout(() => app.modal.show(MyCoolModal, { attr: 'value' }), 0);
+   */
+  show(componentClass: UnsafeModalClass, attrs?: Record<string, unknown>): void;
```

Comment:
in 2.0 we'll probably keep the new `stackModal` argument set to `false` by default (or `true`). So I think we could dro this signature and the deprecation mark. Thoughts?